apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx

```diff
@@ -198,27 +198,33 @@ export const MultiFactorAuthentication: React.FunctionComponent<MfaProps> = (pro
                                 isSuperTenantLogin={ isSuperTenantLogin() }
                                 onEnabledAuthenticatorsUpdated={ handleEnabledAuthenticatorsUpdated }
                                 triggerBackupCodesFlow={ () => setInitBackupCodeFlow(true) }
-                                handleSessionTerminationModalVisibility={ 
-                                    () => setShowSessionTerminationModal(true) 
-                                }
+                                handleSessionTerminationModalVisibility={ () => {
+                                    // Show the session termination modal only if the backup code flow is not activated
+                                    // to stop showing duplicate modals.
+                                    if (isSuperTenantLogin() && isTOTPEnabled && isBackupCodesConfigured) {
+                                        setShowSessionTerminationModal(false);
+                                    }
+
+                                    setShowSessionTerminationModal(true);
```

Comment:
Shouldn't this line fall under else block?

Reply:
Yeah. Missed that. Fixed in https://github.com/wso2/identity-apps/pull/3525/commits/448da508d35f9e086fbf298f09a91713d00c0c52